### DISCUSSION TITLE
How can I get the all entry file name in the loaders? #2090

### DISCUSSION THREAD

**Discussion by lishichao1002** [webpack/webpack#15990] [General]
a.js
```javascript
export const a = 1;
```

main1.js
```javascript
import { a } from './a.js';
console.log(a + 1)
```

main2.js
```javascript
import { a } from './a.js';
console.log(a + 1)
```

main3.js
```javascript
import { a } from './a.js';
console.log(a + 1)
```

webpack.config.js
```javascript
module.exports = {
  entry: {
    main1: './main1.js',
    main2: './main2.js',
    main3 './main3.js',
  },
  module: {
    rules: [
      { test: /\.js$/, use: './my-customer-loader' },
    ],
  },
};
```

my-customer-loader.js
```javascript
module.exports = function(content, map, meta) {
    const filePath = this.resourcePath;
    const fullFilePath = this.resource;
    const entryFiles = // TODO How can I get the all entry file name in the loaders ??
};
```

how can i get all etnry file name in the loaders ?

relative issues https://github.com/webpack/webpack/issues/2090

**Comment by alexander-akait** [comment]
Why you need it? You can do using `this._compilation.entries`. Anyway generate code based is entries often is bad idea, entries can be added late (after loaders) and name can be renamed/hashed/etc

**Reply by lishichao1002** [reply]
hello， the  `this._compilation.entries` is  `all entries`，not `a.js's entries`。

my scene is: 

In order to optimize fmp, I want preload api data，I inject api invoke to html，so api can send as script send time， not wait framework library js download and compile complete ，like below code: 

source code:
vue.js
```javascript
preAxios({
   method: 'get',
   url: '/some/api/url',
}).then(cosnole.log).catch(console.error)
```

build target code:
index.html
```html
<script>
// inject this code in html in order to send xhr early
window.__preAxios__.result_hash_ = axios({ method: get, url: '/some/api/url' })
</script>
<script src="vue.lib.js"></script>
<script src="route.page.js"></script>
```

vue.js
```javascript
// Correlation xhr results
window.__preAxios__.result_hash_.then(cosnole.log).catch(console.error)
```

In order to this result，I need a webpack loader to handler preAxios AST，and in order to inject generated code i need a webpack plugin;  In multi page project, I need to know their relationship , that is ths issuse `how i get all entry file name in the loaders`.

**Reply by alexander-akait** [reply]
You need a plugin to do it, loader design is for transformations, not such graph modifications/changes
